src/discord-cluster-manager/cogs/leaderboard_cog.py

```diff
@@ -251,76 +273,132 @@ async def leaderboard_create(
+                except ValueError:
+                    raise ValueError(
+                        "Invalid date format. Please use YYYY-MM-DD or YYYY-MM-DD HH:MM"
+                    )
 
+            # Ask the user to select GPUs
+            view = GPUSelectionView([gpu.name for gpu in GitHubGPU])
+
+            await interaction.response.send_message(
+                "Please select GPUs for this leaderboard:",
+                view=view,
+                ephemeral=True,
+            )
+
+            # Wait until the user makes a selection
+            await view.wait()
+
+        except ValueError as ve:
+            # Handle invalid date errors
+            await interaction.response.send_message(
+                str(ve),
+                ephemeral=True,
+            )
+
+        # Kind of messy, but separate date try/catch
+        try:
             # Read the template file
             template_content = await reference_code.read()
 
             with self.bot.leaderboard_db as db:
-                print(
-                    leaderboard_name,
-                    type(date_value),
-                    type(template_content.decode("utf-8")),
-                )
-                db.create_leaderboard({
+                err = db.create_leaderboard({
                     "name": leaderboard_name,
                     "deadline": date_value,
                     "reference_code": template_content.decode("utf-8"),
+                    "gpu_types": view.selected_gpus,
                 })
 
-            await interaction.response.send_message(
+                if err:
+                    print(err)
+                    if "duplicate key" in err:
+                        await interaction.followup.send(
+                            f'Error: Tried to create a leaderboard "{leaderboard_name}" that already exists.',
+                            ephemeral=True,
+                        )
+                    else:
+                        # Handle any other errors
+                        await interaction.followup.send(
+                            "Error in leaderboard creation.",
+                            ephemeral=True,
+                        )
+                    return
+
+            await interaction.followup.send(
                 f"Leaderboard '{leaderboard_name}'. Reference code: {reference_code}. Submission deadline: {date_value}",
                 ephemeral=True,
             )
-        except ValueError:
-            await interaction.response.send_message(
-                "Invalid date format. Please use YYYY-MM-DD or YYYY-MM-DD HH:MM",
-                ephemeral=True,
-            )
+
+        except Exception as e:
+            print(str(e))
+            # Handle specific leaderboard name conflict
+            if "already exists" in str(e):
+                await interaction.followup.send(
+                    str(e),
+                    ephemeral=True,
+                )
```

Comment:
How could we get here? Shouldn't this be handled in or near the `"duplicate key" area above?

(I know I'm sounding like a broken record but) we shouldn't send `str(e)` to the Discord user. We should log a message containing `e` and send a generic message to the user like `"Leaderboard already exists"`.

Reply:
I think this has to do with why I ended up re-factoring create_leaderboard to return an error string, because it wasn't being captured in the try except. I've removed this, but also made sure to remove the print errors on Discord and add them to a logger instead.